src/main/java/com/cleanroommc/groovyscript/compat/mods/pyrotech/StoneSawmill.java

```diff
@@ -0,0 +1,186 @@
+    public StoneSawmillRecipe add(String name, IIngredient input, IIngredient blade, ItemStack output, int duration, int woodChips, boolean inherit) {
+        return recipeBuilder()
+                .duration(duration)
+                .woodChips(woodChips)
+                .inherit(inherit)
+                .name(name)
+                .input(input, blade)
+                .output(output)
+                .register();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.REMOVAL, example = @Example("item('minecraft:planks:1')"))
+    public void removeByInput(ItemStack input) {
+        if (GroovyLog.msg("Error removing stone sawmill recipe")
+                .add(IngredientHelper.isEmpty(input), () -> "Input 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (StoneSawmillRecipe recipe : getRegistry()) {
+            if (recipe.getInput().test(input)) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @MethodDescription(type = MethodDescription.Type.REMOVAL, example = @Example("item('pyrotech:material:23')"))
+    public void removeByOutput(IIngredient output) {
+        if (GroovyLog.msg("Error removing stone sawmill recipe")
+                .add(IngredientHelper.isEmpty(output), () -> "Output 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (StoneSawmillRecipe recipe : getRegistry()) {
+            if (output.test(recipe.getOutput())) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @Property(property = "input", comp = @Comp(gte = 1, lt = 3))
+    @Property(property = "output", comp = @Comp(eq = 1))
+    @Property(property = "name")
+    public static class RecipeBuilder extends AbstractRecipeBuilder<StoneSawmillRecipe> {
+
+        @Property(comp = @Comp(gt = 0))
+        private int duration;
+        @Property(value = "wood_chips", comp = @Comp(gte = 0))
+        private int woodChips;
+        @Property
+        private boolean inherit;
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder duration(int time) {
+            this.duration = time;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder woodChips(int woodChips) {
+            this.woodChips = woodChips;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder inherit(boolean inherit) {
+            this.inherit = inherit;
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Pyrotech Stone Sawmill Recipe";
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 1, 2, 1, 1);
```

Comment:
if the second ingredient is set, this also needs to validate that its in `ModuleTechMachineConfig.STONE_SAWMILL.SAWMILL_BLADES`, otherwise it will not be able to be inserted into that slot